File: Check_Entitlement_by_AJS_vbNet/AJS_Lics.vb
Imports RestSharp
Imports System
Imports System.Windows
Imports System.Windows.Forms

Namespace Check_Entitlement_by_AJS
    Friend Module AJS_Lics
        Private _appID As String = "4688092142369552289"
        Private mCheckOnline As Boolean

        Function CheckOnline(ByVal Optional msg As Boolean = True) As Boolean
            If mCheckOnline Then Return True
            Dim _userID As String = ""

            Try
                _userID = TryCast(Autodesk.AutoCAD.ApplicationServices.Application.GetSystemVariable("ONLINEUSERID"), String)
            Catch
                If msg Then MessageBox.Show("Work on AutoCAD > 2014 Only")
                Return False
            End Try

            If _userID.Equals("") Then
                If msg Then MessageBox.Show("Please log-in to Autodesk 360")
                Return False
            End If

            Try
                Dim client As RestClient = New RestClient("https://apps.autodesk.com")
                Dim req As RestRequest = New RestRequest("webservices/checkentitlement")
                req.Method = Method.[GET]
                req.AddParameter("userid", _userID)
                req.AddParameter("appid", _appID)

                System.Net.ServicePointManager.ServerCertificateValidationCallback = AddressOf AcceptAllCertifications
                Dim resp As IRestResponse(Of EntitlementResult) = client.Execute(Of EntitlementResult)(req)
                mCheckOnline = resp.Data IsNot Nothing AndAlso resp.Data.IsValid
            Catch
                If msg Then MessageBox.Show("Cannot connect to the https://apps.autodesk.com")
            End Try

            If mCheckOnline <> True AndAlso msg Then MessageBox.Show("You are not entitled to use this plug-in!")
            Return mCheckOnline
        End Function

        Public Function AcceptAllCertifications(ByVal sender As Object, ByVal certification As System.Security.Cryptography.X509Certificates.X509Certificate, ByVal chain As System.Security.Cryptography.X509Certificates.X509Chain, ByVal sslPolicyErrors As System.Net.Security.SslPolicyErrors) As Boolean
            Return True
        End Function
    End Module
End Namespace

File: Check_Entitlement_by_AJS/AJS_Lics.cs
using RestSharp;
using System;
using System.Windows;

namespace Check_Entitlement_by_AJS
{
    internal static class AJS_Lics
    {
        //Install-Package RestSharp -Version 105.2.3
        //Example by www.lisp.vn

        private static String _appID = "4688092142369552289";
        private static bool mCheckOnline;

        public static bool CheckOnline(bool msg = true)
        {
            if (mCheckOnline) return true;
            String _userID = "";
            try
            {
                _userID = Autodesk.AutoCAD.ApplicationServices.Application.GetSystemVariable("ONLINEUSERID") as String;
            }
            catch
            {
                if (msg)
                    MessageBox.Show("Work on AutoCAD > 2014 Only");
                return false;
            }

            if (_userID.Equals(""))
            {
                if (msg)
                    MessageBox.Show("Please log-in to Autodesk 360");
                return false;
            }

            try
            {
                //check for online entitlement
                RestClient client = new RestClient("https://apps.autodesk.com");
                RestRequest req = new RestRequest("webservices/checkentitlement");
                req.Method = Method.GET;
                req.AddParameter("userid", _userID);
                req.AddParameter("appid", _appID);
                System.Net.ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => true;
                IRestResponse<EntitlementResult> resp = client.Execute<EntitlementResult>(req);

                mCheckOnline = resp.Data != null && resp.Data.IsValid;
            }
            catch
            {
                if (msg)
                    MessageBox.Show("Cannot connect to the https://apps.autodesk.com");
            }

            if (mCheckOnline != true && msg)
                MessageBox.Show("You are not entitled to use this plug-in!");

            //Example by www.lisp.vn
            return mCheckOnline;
        }
    }
}